splice_machine/src/main/java/com/splicemachine/derby/stream/iterator/merge/MergeAntiJoinIterator.java

```diff
@@ -44,6 +44,10 @@ public MergeAntiJoinIterator(Iterator<ExecRow> leftRS,
     public boolean internalHasNext() {
         try {
             while (leftRS.hasNext()) {
+                if (left == null)
+                    left = leftRS.next().getClone();
+                else
+                    left.transfer(leftRS.next());
                 left = leftRS.next();
```

Comment:
This line should be removed. Given that ITs pass I guess this is not covered by any test, could you write a quick one John?